python/tank/authentication/sso_saml2/utils.py

```diff
@@ -11,31 +11,41 @@
 SSO/SAML2 Utility functions.
 """
 
+import time
+
+from tank_vendor import shotgun_api3
+
 from .core.utils import (  # noqa
     get_logger,
     set_logger_parent,
     get_saml_claims_expiration,
+    get_session_expiration,
     _decode_cookies,
     _get_shotgun_user_id,
     _sanitize_http_proxy,
 )
 
 
-def is_sso_enabled_on_site(shotgun_module, url, http_proxy=None):
+# Cache the servers infos foe 30 seconds.
+INFOS_CACHE_TIMEOUT = 30
+INFOS_CACHE = {}
```

Comment:
this kind of cache can cause issues in bootstrap core swapping scenarios (right @jfboismenu ? ). Is there a way we could make it part of the state instead of global?

Reply:
Not really actually. It just means that after a core swap the cache will go away and will need to be repopulated.

Reply:
Spoke to JF, we both agree that if the cache is trashed it is a minor inconvenience to have to re-do the system call.

Reply:
if this absolutely has to be expressed by a global state - and thus affected (even in a harmless/minor way) by bootstrap, please add a comment just to be clear. I am not just thinking about the current behaviour here, but imagine someone making a future change to the code - the core swap is a headache - which we have to live with for the time being - because a seemingly harmless change to a global state like this may have weird and unexpected sideffects. So ideal case: we always try to avoid introducing global state in toolkit if at all possible, second best option - add a comment to point it out to future maintainers of the code! thanks! ❤️

Reply:
Added a nice comment to describe INFOS_CACHE and the effect of a core swap.